src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -337,8 +337,9 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 
 	private showTaskNotification(): void {
 		const storageKey = 'workbench.tasks.ranTaskBefore';
-		if (!this.storageService.get(storageKey) && this.contextService.getWorkspace()) {
-			const fileName = path.relative(this.contextService.getWorkspace().resource.toString(), this.resource.toString());
+		const ignoreKey = 'workbench.tasks.ignoreTaskNotification';
+		if (!this.storageService.get(ignoreKey) && (!this.storageService.get(storageKey) && this.contextService.getWorkspace2())) {
```

Comment:
You might want to check that getWorkspace2().roots[0] is defined as well. It looks like if a folder isn't open, then getWorkspace2 should return null and this would be fine. But since this is a new API subject to change then we should be conservative here. Also this line has unnecessary `(...)`